File: backend/src/eSamadhaan.Domain/Entities/Feedback.cs
namespace eSamadhaan.Domain.Entities;

public class Feedback
{
    public int Id { get; set; }
    public int GrievanceId { get; set; }
    public int Rating { get; set; }
    public string Comment { get; set; } = string.Empty;
    public DateTime SubmittedAt { get; set; }

    // Navigation properties
    public Grievance Grievance { get; set; } = null!;
}

File: frontend/esamadhaan-ui/src/app/models/feedback.ts
// Feedback Interfaces

// Request DTOs
export interface CreateFeedbackRequest {
  rating: number; // 1-5
  comment?: string | null;
}

// Response DTOs
export interface FeedbackResponseDto {
  id: number;
  grievanceId: number;
  grievanceNumber: string;
  rating: number; // 1-5
  comment: string | null;
  submittedAt: string; // ISO 8601 DateTime
}

